src/actions/dataActions.tsx

```diff
@@ -810,12 +807,58 @@ export function _configureServers(
 
 ////////////////////////////////////////////////////////////////////////////////
 
-export function syncWithServer() {
-  return (dispatch: Dispatch) => {
+export function syncWithServer(store: Store) {
+  return (dispatch: Dispatch, getState: () => AppState) => {
     dispatch(updateServerInfo() as unknown as Action);
     dispatch(updateDatasets() as unknown as Action);
     dispatch(updateExpressionCapabilities() as unknown as Action);
     dispatch(updateColorBars() as unknown as Action);
+
+    const apiServer = selectedServerSelector(getState());
+    initializeContributions({
+      hostStore: newHostStore(store),
+      logging: { enabled: import.meta.env.DEV },
+      api: { serverUrl: apiServer.url, endpointName: "viewer/ext" },
+    });
+  };
+}
+
+function newHostStore(store: Store): StoreApi<AppState> & {
+  _initialState: AppState;
+  _prevState: AppState;
+} {
+  return {
+    _initialState: store.getState(),
+    getInitialState(): AppState {
+      // noinspection JSPotentiallyInvalidUsageOfThis
+      return this._initialState;
+    },
+    getState(): AppState {
+      return store.getState();
+    },
+    setState(
+      _state:
+        | AppState
+        | Partial<AppState>
+        | ((state: AppState) => AppState | Partial<AppState>),
+      _replace?: boolean,
+    ): void {
+      throw new Error(
+        "Changing the host state from contributions is not yet supported",
+      );
+    },
+    _prevState: store.getState(),
```

Comment:
here both `_initialState` and `_prevState` are assigned the same state. Can you explain why?